file-formats/saia/type/zif_aff_saia_v1.intf.abap

```diff
@@ -0,0 +1,142 @@
+      "! <p class="shorttext">Supported Resource Types</p>
+      "! Supported resource types
+      supported_resource_types    TYPE ty_resource_types,
+    END OF ty_filters.
+
+  "! <p class="shorttext">ABAP Class Name</p>
+  "! ABAP class name
+  TYPES ty_class_name TYPE c LENGTH 30.
+
+  TYPES:
+    "! <p class="shorttext">ADT IDE Action</p>
+    "! ADT IDE Action
+    "! $required
+    BEGIN OF ty_adt_saia_object,
+      "! <p class="shorttext">Unique Id of the Action</p>
+      "! Unique action id - it should be human readable and reveal the intention of the action
+      "! Action id is case insensitive
+      "! TODO: length restriction, check how long it should be, su22 variant restrictions???
+      "! $required
+      action_id                      TYPE string,
+      "! <p class="shorttext">Action Title</p>
+      "! Action title for the action
+      "! $required
+      title                          TYPE string,
+      "! <p class="shorttext">Action Description</p>
+      "! What is the action doing and how can it be used
+      "! $required
+      description                    TYPE string,
+      "! <p class="shorttext">Action Filter(In development)</p>
+      "! Filter for application of action according to specific object type or object type groups
+      "! TODO: new filter mechanis currently under development -> all filters stored in a string on database
+      "! $required
+      filters                        TYPE ty_filters,
+      "! <p class="shorttext">Action Handler Class</p>
+      "! Implementation class for handling the action input. Needs to implement interface  {@link if_aia_action }.
+      "! $required
+      implementation_class           TYPE ty_class_name,
+      "! <p class="shorttext">Action Input UI Class</p>
+      "! The ABAP class implementing the server-driven UI input configuration.
+      "! Needs to implement interface {@link IF_AIA_SD_ACTION_INPUT }.
+      "! $required
+      input_ui_configuration_class   TYPE ty_class_name,
+      "! <p class="shorttext">Run Action Dialog Visibility</p>
+      "! Flag to indicate that the action shouldn't be shown in the action selection dialog
+      "!   where all available actions are listed.
+      "! TODO: put in a structure: options -> all options stored in a string in database
+      "! $required
+      dont_show_in_run_action_dialog TYPE abap_bool,
+    END OF ty_adt_saia_object.
+
+  TYPES:
+    "! <p class="shorttext">ADT IDE Action</p>
+    "! ADT IDE Action (SAIA) v1
+    BEGIN OF ty_main,
+      "! <p class="shorttext">Format Version</p>
+      "! Format version
+      "! $required
+      format_version TYPE zif_aff_types_v1=>ty_format_version,
+      "! <p class="shorttext">Header</p>
+      "! Header
+      "! $required
+      header         TYPE zif_aff_types_v1=>ty_header_100,
+      "! <p class="shorttext">ADT IDE Action</p>
+      "! ADT IDE Action
+      "! $required
+      adt_ide_action TYPE ty_adt_saia_object,
```

Comment:
I'm not sure about this field. The whole file format describes the ADT IDE Action and not only this component here. So maybe something like action_properties would make sense. (or even our generic term general_information)